docs/src/proposals/optimistic-transaction-propagation-signal.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Optimistic Transaction Propagation Signal
+---
+
+## Current Retransmit behavior
+
+The retransmission tree currently considers:
+1. epoch staked nodes
+2. tvu peers (filtered by contact info and shred version)
+3. current validator
+
+concatenating (1), (2), and (3)
+deduplicating this list of entries by pubkey favoring entries with contact info
+filtering this list by entries with contact info
+
+This list is then is randomly shuffled by stake weight.
+
+Shreds are then retransmitted to up to FANOUT neighbors and up to FANOUT
+children.
+
+## Deterministic retransmission tree
```

Comment:
I do already have some draft code related to this part. That was actually the motivation of adding `ClusterNodes` and `ClusterNodesCache`, so that I can make this change. I can tidy that out and send that as a pull-request.

Reply:
Cool, if you already have changes which would separate the retransmission into a deterministic set from the epoch staked nodes that sounds great.